### PR TITLE
fix(pci-rancher): Fetching error on PCI Project ID

### PR DESCRIPTION
**Pull request by Tsiorifamonjena**
ref: TAPC-1284

| Question         | Answer
| ---------------- | ---
| Branch?          | `develop`
| Bug fix?         | yes
| New feature?     | no
| Breaking change? | no
| Tickets          | TAPC-1284
| License          | BSD 3-Clause

- [x] Try to keep pull requests small so they can be easily reviewed.
- [x] Commits are signed-off
- [ ] Only FR translations have been updated
- [ ] Branch is up-to-date with target branch
- [x] Lint has passed locally
- [x] Standalone app was ran and tested locally
- [ ] Ticket reference is mentioned in linked commits (internal only)
- [ ] Breaking change is mentioned in relevant commits

## Description

Check directly if the pci project ID exists when arriving on one of the rancher pages to avoid fetching unnecessary calls on mounting the rest of the pages

### CODE REVIEW

File: packages/manager/apps/pci-rancher/src/hooks/useRancher.tsx [hunk 1]
```diff
@@ -18,13 +18,13 @@ export const useRancher = ({
 } = {}) => {
   const { projectId, rancherId } = useParams();
   return useQuery<RancherService, ErrorResponse>({
-    queryKey: ['project', projectId, 'rancher', rancherId],
+    queryKey: ['projectId', projectId, 'rancher', rancherId],
```
Comment (fredericvilcot) [inline packages/manager/apps/pci-rancher/src/hooks/useRancher.tsx:21]: If `project` has been switched to `projectId`, I think we should also harmonize it with `rancher` ;)

**Review by anooparveti — CHANGES_REQUESTED** [review]
Commits are not signed-off. Can you please sign-off the commits?

File: packages/manager/apps/pci-rancher/src/hooks/useRancher.tsx (hunk 1 above)
Comment (Tsiorifamonjena) [reply packages/manager/apps/pci-rancher/src/hooks/useRancher.tsx]: I switch to `projectId` to fix a conflict cache key from `useProject` hook. But yes I agree with you, I change it so to harmonize

**Review by fredericvilcot — APPROVED** [review]
Great 👌

**Review by anooparveti — CHANGES_REQUESTED** [review]
There is an ongoing work regarding the movement of `useProject` hook from `manager-components` to a different folder. #12267 
So, better to synchronise this PR with the base as #12267

**Comment by Tsiorifamonjena** [comment]
> There is an ongoing work regarding the movement of `useProject` hook from `manager-components` to a different folder. #12267 So, better to synchronise this PR with the base as #12267

@anooparveti the PR is not yet merged, it is better to wait or what can I do ? I don't know if cherry-pick will work

**Comment by anooparveti** [comment]
> There is an ongoing work regarding the movement of `useProject` hook from `manager-components` to a different folder. #12267 So, better to synchronise this PR with the base as #12267

This comment will be handled by TAPC-1600

**Review by anooparveti — APPROVED** [review]

**Pull request merged**